core/templates/pages/contributor-dashboard-page/services/contribution-and-review-backend-api.service.ts

```diff
@@ -0,0 +1,165 @@
+}
+
+interface UpdateTranslationData {
+  'translation_html': string;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ContributionAndReviewBackendApiService {
+  private _SUBMITTED_SUGGESTION_LIST_HANDLER_URL = (
+    '/getsubmittedsuggestions/<target_type>/<suggestion_type>');
+  private _REVIEWABLE_SUGGESTIONS_HANDLER_URL = (
+    '/getreviewablesuggestions/<target_type>/<suggestion_type>');
+  private _SUGGESTION_TO_EXPLORATION_ACTION_HANDLER_URL = (
+    '/suggestionactionhandler/exploration/<exp_id>/<suggestion_id>');
+  private _SUGGESTION_TO_SKILL_ACTION_HANDLER_URL = (
+    '/suggestionactionhandler/skill/<skill_id>/<suggestion_id>');
+  private _UPDATE_TRANSLATION_HANDLER_URL = (
+    '/updatetranslationsuggestionhandler/<suggestion_id>');
+  private _UPDATE_QUESTION_HANDLER_URL = (
+    '/updatequestionsuggestionhandler/<suggestion_id>');
+
+  private _SUBMITTED_QUESTION_SUGGESTIONS = (
+    'SUBMITTED_QUESTION_SUGGESTIONS');
+  private _REVIEWABLE_QUESTION_SUGGESTIONS = (
+    'REVIEWABLE_QUESTION_SUGGESTIONS');
+  private _SUBMITTED_TRANSLATION_SUGGESTIONS = (
+    'SUBMITTED_TRANSLATION_SUGGESTIONS');
+  private _REVIEWABLE_TRANSLATION_SUGGESTIONS = (
+    'REVIEWABLE_TRANSLATION_SUGGESTIONS');
+
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService
+  ) {}
+
+  async fetchSuggestionsAsync(
+      fetchType: string
+  ): Promise<FetchSuggestionsResponse> {
+    if (fetchType === this._SUBMITTED_QUESTION_SUGGESTIONS) {
+      return this.fetchSubmittedSuggestionsAsync('skill', 'add_question');
+    }
+    if (fetchType === this._SUBMITTED_TRANSLATION_SUGGESTIONS) {
+      return this.fetchSubmittedSuggestionsAsync(
+        'exploration', 'translate_content');
+    }
+    if (fetchType === this._REVIEWABLE_QUESTION_SUGGESTIONS) {
+      return this.fetchReviewableSuggestionsAsync('skill', 'add_question');
+    }
+    if (fetchType === this._REVIEWABLE_TRANSLATION_SUGGESTIONS) {
+      return this.fetchReviewableSuggestionsAsync(
+        'exploration', 'translate_content');
+    }
+  }
+
+  async fetchSubmittedSuggestionsAsync(
+      targetType: string,
+      suggestionType: string
+  ): Promise<FetchSuggestionsResponse> {
+    let url = this.urlInterpolationService.interpolateUrl(
+      this._SUBMITTED_SUGGESTION_LIST_HANDLER_URL, {
+        target_type: targetType,
+        suggestion_type: suggestionType
+      }
+    );
+    return this.http.get<FetchSuggestionsResponse>(url).toPromise();
+  }
+
+  async fetchReviewableSuggestionsAsync(
+      targetType: string,
+      suggestionType: string
+  ): Promise<FetchSuggestionsResponse> {
+    let url = this.urlInterpolationService.interpolateUrl(
+      this._REVIEWABLE_SUGGESTIONS_HANDLER_URL, {
+        target_type: targetType,
+        suggestion_type: suggestionType
+      }
+    );
+    return this.http.get<FetchSuggestionsResponse>(url).toPromise();
+  }
+
+  async resolveToExplorationAsync(
```

Comment:
This request "applies" an action (accept, reject) to an exploration suggestion, so "resolveToExploration" doesn't seem quite right here. The "actions" are also only related to completing a suggestion review, so maybe something like `reviewExplorationSuggestionAsync`. Likewise below.

Reply:
done